dnf/cli/commands/history.py

```diff
@@ -167,24 +165,10 @@ def _hcmd_redo(self, extcmds):
 
         # TODO(dmach): refactor store/replay code to avoid using the temporary file
         data = serialize_transaction(old)
-        tmp_dir = tempfile.mkdtemp(prefix="dnf_history_redo_")
-        filename = os.path.join(tmp_dir, "transaction.json")
-        try:
-            # store the transaction into a json file that will be loaded in the following step
-            with open(filename, "w") as f:
-                json.dump(data, f, indent=4, sort_keys=True)
-                f.write("\n")
-
-            # replay the transaction from file
-            self.base.read_comps(arch_filter=True)
-            self.replay = TransactionReplay(self.base, filename, ignore_installed=True, ignore_extras=True, skip_unavailable=self.opts.skip_unavailable)
-            self.replay.run()
-        finally:
-            # cleanup, remove the json with stored transaction
-            try:
-                shutil.rmtree(tmp_dir)
-            except:
-                pass
```

Comment:
Not a blocker or anything, but if you squashed this commit with the respective previous commits to drop the usage of the temp file from git history, I don't think it would be particularly hard and an interesting excersize :slightly_smiling_face: but up to you.